src/Infra/MySQL/MySQLItemRepository.ts

```diff
@@ -100,4 +108,24 @@ export class MySQLItemRepository extends Repository<Item> implements ItemReposit
       )
       .execute()
   }
+
+  async markAsDeleted(item: Item): Promise<void> {
+    const timestamp = this.timer.getTimestampInMicroseconds()
+    await this.createQueryBuilder('item')
+      .update()
+      .set({
+        deleted: true,
+        content: null,
+        encItemKey: null,
+        authHash: null,
+        updatedAtTimestamp: timestamp,
```

Comment:
@karolsojko can we use [listeners](https://github.com/typeorm/typeorm/blob/master/docs/listeners-and-subscribers.md) to automatically update timestamps on/before save? I think it's dangerous to be manually updating timestamps in various locations.

Reply:
I'm not a fan of listener since they add automagic to the picture - Repositories (like `MySQLItemRepository`) are the only place where data modifications happen. So it's not like there are various locations updating the timestamps. To be honest there was one place only uptil now I think

Reply:
So what should we do now that we have introduced multiple locations? Is it perhaps this PR that needs to be refactored somehow? What do you recommend?